test/inductor/test_compiled_optimizers.py

```diff
@@ -482,6 +482,76 @@ def fn():
         self.assertTrue(p_ref() is None)
         gc.enable()
 
+    # Basic shampoo test to verify we support compiling the various ops without error
+    @requires_cuda
+    def test_basic_shampoo(self):
+        param_buf = torch.rand((1024, 128))
+        param_buf_c = param_buf.clone().detach()
+
+        params_c = [param_buf_c[0:512, :], param_buf_c[512:, :]]
```

Comment:
One thing to call out is that depending on how closely we want to model the shampoo codepath, there are usually:

(1) 100+ params that all alias other (there are only 2 here)

(2) each param is not always contiguous (they're all contiguous here).

Maybe (1) fine to relax for this test, since we'll have other more dedicated unit tests for non_overlapping lots of aliases that assert there aren't too many guards. But we might want the representative test here to more accurately replicate the noncontiguous pattern of the params?

Reply:
Yeah I'll make it non-contiguous, that sounds fine. We will have testing on diffs which will run the full shampoo correctness, so I'm okay with it not 100% matching.